src/acquisition/rvdss/utils.py

```diff
@@ -0,0 +1,199 @@
+import requests
+import pandas as pd
+import io
+import regex as re
+from epiweeks import Week
+from datetime import datetime
+import math
+
+def abbreviate_virus(full_name):
+    lowercase=full_name.lower()
+    
+    if any(name in lowercase for name in ["parainfluenza","para","piv"]):
+        if "hpiv" not in lowercase:
+            abbrev = re.sub("parainfluenza|para|piv","hpiv",lowercase)
+        else:
+            abbrev = lowercase
+    elif any(name in lowercase for name in ["adenovirus","adeno"]):
+        abbrev =  re.sub("adenovirus|adeno","adv",lowercase)
+    elif "human metapneumovirus" in lowercase:
+        abbrev =  re.sub("human metapneumovirus","hmpv",lowercase)
+    elif any(name in lowercase for name in ["enterovirus/rhinovirus","rhinovirus","rhv","entero/rhino","rhino","ev/rv","evrv"]):
+        abbrev = re.sub("enterovirus/rhinovirus|rhinovirus|rhv|entero/rhino|rhino|ev/rv|evrv","ev_rv",lowercase)
+    elif any(name in lowercase for name in ["coronavirus","coron","coro"]):
+        abbrev = re.sub("coronavirus|coron|coro","hcov",lowercase)
+    elif "respiratory syncytial virus" in lowercase:
+        abbrev = re.sub("respiratory syncytial virus","rsv",lowercase)
+    elif "influenza" in lowercase:
+        abbrev = re.sub("influenza","flu",lowercase)       
+    elif "sarscov2" in lowercase:
+        abbrev = re.sub("sarscov2","sars-cov-2",lowercase) 
+    else:
+        abbrev=lowercase
+    return(abbrev)
+
+def abbreviate_geo(full_name):
+    lowercase=full_name.lower()
+    
+    if "newfoundland" in lowercase:
+        abbrev =  "nl"
+    elif "prince edward island" in lowercase:
+        abbrev =  "pe"
+    elif "nova scotia" in lowercase:
+        abbrev =  "ns"
+    elif "new brunswick" in lowercase:
+        abbrev =  "nb"
+    elif "nova scotia" in lowercase:
+        abbrev =  "ns"     
+    elif re.match('|'.join(("^québec$", "province of québec","quebec")),lowercase):
+        abbrev = "qc"  
+    elif re.match('|'.join(("^ontario$", "province of ontario")),lowercase):
+        abbrev =  "on"
+    elif "manitoba" in lowercase:
+        abbrev =  "mb"
+    elif "saskatchewan" in lowercase:
+        abbrev =  "sk"
+    elif "alberta" in lowercase:
+        abbrev =  "ab"
+    elif "british columbia" in lowercase:
+        abbrev =  "bc"
+    elif "yukon" in lowercase:
+        abbrev =  "yk"
+    elif "northwest territories" in lowercase:
+        abbrev =  "nt"
+    elif "nunavut" in lowercase:
+        abbrev =  "nu"
+    elif re.match("canada|can",lowercase):
+        abbrev = "ca" 
+    elif re.match(r"^at\b",lowercase):
+        abbrev = "atlantic" 
+    elif "pr" in lowercase:
+        abbrev = "prairies" 
+    elif "terr" in lowercase:
+        abbrev = "territories" 
+    else:
+        abbrev=lowercase
+    return(abbrev)
+
+def create_geo_types(geo,default_geo):
+    regions = ['atlantic','atl','province of québec','québec','qc','province of ontario','ontario','on',
```

Comment:
question: I know we [previously decided](https://delphi-org.slack.com/archives/C078XQGDHJR/p1724969838674169) that we'd mix 2-letter province abbreviations and full region names when returning regional data (since some provinces are their own region), but I'm questioning that now. Should we just mix 2-letter province abbrevs with longer regional abbreviations (like `atl`, `terr`, `pr` listed here)?

Reply:
Decided in discussions with Dan that, since regions don't have standard abbreviations, we should just list out their full names.